src/org/labkey/test/components/ui/grids/QueryGrid.java

```diff
@@ -721,6 +732,16 @@ public WebElement panelHeader()
             return Locator.xpath("preceding-sibling::div[contains(@class,'panel-heading')]").findWhenNeeded(this);
         }
 
+        final BootstrapMenu chartsMenu = new MultiMenu.MultiMenuFinder(getDriver()).withText("Charts").findWhenNeeded(this);
+
+        final WebElement chartPanel = Locator.byClass("chart-panel").refindWhenNeeded(this);
```

Comment:
FYI if this grid is going to be used to show more than a couple of charts, 'refindWhenNeeded' doesn't re-find indefinitely (I think this is to avoid endless looping re-tries).

Reply:
Yeah, in theory we could write a test case that does something like:

- Navigate to grid
- Click chart menu, select chart
- Verify chart output
- Click chart menu, select chart
- rinse, repeat

Which is why I went for refindWhenNeeded because I thought that was how we avoided "stale" element issues. Is there a better solution?

Reply:
To support re-use of the page to show multiple charts or reports, you could find the report new instead of caching a refindingElement.  So, instead of 
`final WebElement chartPanel = Locator.byClass("chart-panel").refindWhenNeeded(this);`

you could do this
`WebElement chartPanel()
    {
        return  Locator.byClass("chart-panel").waitForElement(2000, this);
    }`